src/marshmallow/fields.py

```diff
@@ -802,7 +802,7 @@ def _validated(self, value):
             return None
         try:
             return self._format_num(value)
-        except (TypeError, ValueError):
+        except (TypeError, ValueError, OverflowError):
```

Comment:
It might be good to handle this separately, with a new failure type, e.g. `'overflow'` or `'too_large'`.